### DISCUSSION TITLE
Can repeated messages be suppressed/joined together?

### DISCUSSION THREAD

**Discussion by dotfrag** [folke/noice.nvim#863] [Q&A] [answered]
I have used the following config to move the `Pattern not found` error messages to `mini` view:

```lua
          {
            filter = {
              event = "msg_show",
              find = "E486",
            },
            view = "mini",
          },
```

And it's working fine, but when spamming `n`/`N`, the messages are printed again. Can I make repeating (same) messages not appear multiple times? The ideal scenario would be for the last repeated message to replace the previous, and extend the timeout before disappearing.

I couldn't find anything in readme or discussions, any pointers are welcome.

Side question: As soon as a message disappears (times out), triggering the same msg won't appear again, even though it's technically a new message. Can this be changed?

**Comment by tranbinh1604** [comment]
try this, work for me

```
{ -- messsages handle by mini
        filter = {
          event = "msg_show",
          any = {
            { find = "%dL" },
            { find = "E486: Pattern not found:.*" },
          },
        },
        view = "mini",
      }
```

**Reply by dotfrag** [reply]
Sorry I did not understand how this answers the question.

**Reply by tranbinh1604** [reply]
> Sorry I did not understand how this answers the question.

Just put it in noice.lua under the routes block, below is mind:

```
return {
  "folke/noice.nvim",
  opts = {
    routes = {
      { -- messages to skip
        filter = {
          event = "notify",
          find = "No information available",
        },
        opts = {
          skip = true,
        },
      },
      { -- messsages handle by mini
        filter = {
          event = "msg_show",
          any = {
            { find = "%dL" },
            { find = "E486: Pattern not found:.*" },
            { find = "E21: Cannot make changes, 'modifiable' is off.*" },
          },
        },
        view = "mini",
      },
    },
    presets = {
      lsp_doc_border = true,
    },
  },
}
```

**Reply by dotfrag** [reply]
That doesn't override consequent messages, it stacks them on top of each other.

EDIT: screenshot, I want to able to make repetitive notifications to override each other, not stack on top of each one.

![2024-06-16T16:41:15,934384043+03:00](https://github.com/folke/noice.nvim/assets/17456867/285a86b1-f8b1-4288-a051-cb2ab53affbe)

**Reply by tranbinh1604** [reply]
You can change to skip = true like above I have, if you want to override this messages, as far as I know, noice only take the messages from nvim and display them, then you will need to play with nvim messages. With me it’s totally ok to push this kind off message to the right bottom corner

**Reply by dotfrag** [reply]
I appreciate you taking the time to help me, but I might have worded my question wrong. I do not want to suppress (skip) all the messages. I want repeating messages with the same text to not appear multiple times, only once.

**Comment by dotfrag** [comment]
@folke can you please share your lights on this one?

TLDR:
- I would like for repeating (same) notification to override each other (instead of stacking on top of each other), is this possible?
- When a notification times out and goes away, if the same notification (same text) is triggered again, it won't appear. Is this a limitation in nvim messaging system or a noice problem/config?

**Reply by folke** [reply, marked as answer]
not possible right now

**Reply by dotfrag** [reply]
Thanks, any idea for the second point?
